src/containers/BooksForm.js

```diff
@@ -1,20 +1,82 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { createBook } from '../actions/index';
 
-const BooksForm = () => {
-  const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+class BooksForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: '',
+    };
+    this.baseState = this.state;
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
 
-  return (
-    <form>
-      <label htmlFor="title">
-        Title
-        <input id="title" type="text" name="title" />
-      </label>
-      <select id="category">
-        {categories.map(e => <option key={e} value={e}>{e}</option>)}
-      </select>
-      <button type="submit">Submit</button>
-    </form>
-  );
-};
+  async handleChange(e, field) {
+    await this.setState({
+      [field]: e.target.value,
+    });
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    const { createBook } = this.props;
+    createBook(this.state);
+    this.setState(this.baseState);
+  }
+
+  render() {
+    const categories = [
+      'Action',
+      'Biography',
+      'History',
+      'Horror',
+      'Kids',
+      'Learning',
+      'Sci-Fi',
+    ];
+    const { title, category } = this.state;
+    return (
+      <form onSubmit={event => this.handleSubmit(event)}>
+        <label htmlFor="title">
+          Title
+          <input
+            id="title"
+            type="text"
+            name="title"
+            title={title}
+            onChange={event => this.handleChange(event, 'title')}
+          />
+        </label>
+        <select
+          id="category"
+          name="category"
+          category={category}
```

Comment:
Did you mean `value`? You must set the `value` attribute to the corresponding value in the `state`, otherwise they are not tied together.

<details>
 <summary>Hint</summary>
 
```suggestion
          value={category}
```

</details>

https://reactjs.org/docs/forms.html